components/common/hooks/useDelinquencySubscription.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {useSelector} from 'react-redux';
+
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {GlobalState} from 'types/store';
+import {isSystemAdmin} from 'mattermost-redux/utils/user_utils';
+
+import useGetSubscription, {assertIsSubscription} from './useGetSubscription';
+
+interface UseDelinquencySubscriptionArgs {
+    checkAdmin: boolean;
+}
+
+export const useDelinquencySubscription = ({checkAdmin}: UseDelinquencySubscriptionArgs) => {
+    const subscription = useGetSubscription();
+    const currentUser = useSelector((state: GlobalState) =>
+        getCurrentUser(state),
+    );
+
+    const isDelinquencySubscription = (): boolean => {
+        if (!subscription) {
+            return false;
+        }
+
+        if (!checkAdmin === isSystemAdmin(currentUser.roles)) {
+            return false;
+        }
```

Comment:
Seems like this admin check will probably make more sense outside of this hook as it doesn't relate to the delinquent subscription. 0/5 but it may be cleaner to just add the admin check directly in the feature you want to only show to admins and keep separation of concerns

Reply:
I agree. When I did it I wasn't totally sure that will easy to understand, as it was hard for me even. I'm going to revert this and let each component do it.